src/main/java/com/karankumar/bookproject/backend/service/AuthorService.java

```diff
@@ -48,20 +47,16 @@ public Author findById(Long id) {
     }
 
     @Override
-    public void save(Author author) {
-        if (author != null) {
-            List<Long> matchingAuthors = findAll().stream()
-                .map(BaseEntity::getId)
-                .filter(id -> id.equals(author.getId())).collect(Collectors.toList());
-            if (matchingAuthors.size() == 1) {
-                // if an author with the same ID exists, set this incoming author ID to null so that
-                // a new row in the table is made rather than updating the row that has the same ID
-                LOGGER.log(Level.INFO, "Matching authorIds: " + matchingAuthors);
-                author.removeId();
+    public void save(@NonNull Author author) {
+        if (author.getId() != null) {
+            Author existingAuthor = findById(author.getId());
+            if (existingAuthor != null) {
+                LOGGER.log(Level.INFO, "Matching authorIds: " + existingAuthor.getId());
+                author = new Author(author.getFirstName(), author.getLastName());
```

Comment:
To me it looks much better than the previous implementation, @knjk04 what do you think?